### PR TITLE
[Mellanox] Use the explicit mst device of SPC during fw-upgrade

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

mlxfwmanager on smartwitch will show more than 1 device. Thus update the mlnx-fw-upgrade script to operate explicitly on Spectrum PCI device.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

1) Used the `mlxfwmanager --query-format XML` command to get the output in XML format and used xmlstarlet cmd to parse the output
Example cmd output:
```
<Devices>
    <Device pciName="/dev/mst/mt41692_pciconf0" type="BlueField3" psid="MT_0000001139" partNumber="BF3-COM-DPU_QP_Ax">
      <Versions>
        <FW current="32.41.1000" available="N/A"/>
      </Versions>
      <MACs Base_Mac="N/A" />
      <GUIDs Base_Guid="N/A" />
      <Status>No matching image found</Status>
      <Description>NVIDIA BlueField-3 DPU based COM express; IPN QP</Description>
    </Device>
    <Device pciName="/dev/mst/mt53104_pci_cr0" type="Spectrum3" psid="MT_0000001189" partNumber="MSN4280-WS2RXO_Ax">
      <Versions>
        <FW current="30_2012_4038-006" available="N/A"/>
      </Versions>
      <MACs Base_Mac="N/A" />
      <Status>No matching image found</Status>
      <Description>NVIDIA Spectrum-3 based 400GbE 2U Ethernet Smart-Switch with ONIE; 28 ports QSFP-DD; 4 DPUs; 2 power supplies (AC); x86 CPU - 16 Cores; Secure-boot Capable; standard depth; C2P airflow; Tool-less Rail Kit</Description>
    </Device>
</Devices>
```
2) Exclusively operate on the PCI device inferred from the output to Wait, fetch psid and Current FW version
2) Use awk for a reliable extraction of Available FW output

#### How to verify it

Verify the script if it upgrades FW properly during switch init and during reboot. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by oleksandrivantsiv — APPROVED** [review]

**Review by dgsudharsan — APPROVED** [review]

File: files/build_templates/sonic_debian_extension.j2 [hunk 1]
```diff
@@ -1049,8 +1049,8 @@ sudo rm -rf $FILESYSTEM_ROOT/$MLNX_SONIC_PLATFORM_PY3_WHEEL_NAME
 sudo cp platform/mellanox/nv-syncd-shared/nv-syncd-shared.service $FILESYSTEM_ROOT_USR_LIB_SYSTEMD_SYSTEM/
 sudo LANG=C chroot $FILESYSTEM_ROOT systemctl enable nv-syncd-shared
 
-# Install minicom package
-sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install minicom
+# Install required packages specific for mellanox platform
+sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install minicom xmlstarlet
```
Comment (lguohan) [inline files/build_templates/sonic_debian_extension.j2:1053]: minicom is not specific for mellanox platform,

the comments is not accurate.

Comment (vivekrnv) [reply files/build_templates/sonic_debian_extension.j2:1053]: This logic is under. So these packages are only specific to mellanox

{% if sonic_asic_platform == "mellanox" %}
{$ endif %}

**Pull request merged**